### PR TITLE
fix github issues

### LINKED ISSUE #2546: [BUG]: predicting failing in multiclass - edge case
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the develop branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@develop).

### Issue Description

This looks like some kind of edge case. see below to reproduce the error.

### Reproducible Example

```python
from pycaret.datasets import get_data
data = get_data('questions')

from pycaret.classification import *
s = setup(data, target = 'Next_Question', session_id = 123)

dt = create_model('dt')
predict_model(dt);
```

### Expected Behavior

It should work seamlessly.

### Actual Results

```python-traceback
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Input In [13], in <cell line: 1>()
----> 1 predict_model(dt)

File ~\anaconda3\envs\pycaret-develop\lib\site-packages\pycaret\classification\functional.py:2078, in predict_model(estimator, data, probability_threshold, encoded_labels, raw_score, drift_report, round, verbose)
   2075 if experiment is None:
   2076     experiment = _EXPERIMENT_CLASS()
-> 2078 return experiment.predict_model(
   2079     estimator=estimator,
   2080     data=data,
   2081     probability_threshold=probability_threshold,
   2082     encoded_labels=encoded_labels,
   2083     raw_score=raw_score,
   2084     drift_report=drift_report,
   2085     round=round,
   2086     verbose=verbose,
   2087 )

File ~\anaconda3\envs\pycaret-develop\lib\site-packages\pycaret\classification\oop.py:2189, in ClassificationExperiment.predict_model(self, estimator, data, probability_threshold, encoded_labels, raw_score, drift_report, round, verbose)
   2116 def predict_model(
   2117     self,
   2118     estimator,
   (...)
   2125     verbose: bool = True,
   2126 ) -> pd.DataFrame:
   2128     """
   2129     This function predicts ``Label`` and ``Score`` (probability of predicted
   2130     class) using a trained model. When ``data`` is None, it predicts label and
   (...)
   2186 
   2187     """
-> 2189     return super().predict_model(
   2190         estimator=estimator,
   2191         data=data,
   2192         probability_threshold=probability_threshold,
   2193         encoded_labels=encoded_labels,
   2194         raw_score=raw_score,
   2195         drift_report=drift_report,
   2196         round=round,
   2197         verbose=verbose,
   2198     )

File ~\anaconda3\envs\pycaret-develop\lib\site-packages\pycaret\internal\pycaret_experiment\supervised_experiment.py:4944, in _SupervisedExperiment.predict_model(self, estimator, data, probability_threshold, encoded_labels, raw_score, drift_report, round, verbose, ml_usecase, display, preprocess)
   4942 pred = pred.astype(int)
   4943 if not raw_score:
-> 4944     score = [s[pred[i]] for i, s in enumerate(score)]
   4945 try:
   4946     score = pd.DataFrame(score, index=X_test_.index)

File ~\anaconda3\envs\pycaret-develop\lib\site-packages\pycaret\internal\pycaret_experiment\supervised_experiment.py:4944, in <listcomp>(.0)
   4942 pred = pred.astype(int)
   4943 if not raw_score:
-> 4944     score = [s[pred[i]] for i, s in enumerate(score)]
   4945 try:
   4946     score = pd.DataFrame(score, index=X_test_.index)

IndexError: index 3 is out of bounds for axis 0 with size 3
```

### Installed Versions

3.0

### PR DESCRIPTION
**Pull request by tvdboom**
## Related Issue or bug

Info about Issue or bug

Closes #2546, #2538, #2535, #2477, #2492

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: pycaret/classification/oop.py [hunk 1]
```diff
@@ -252,7 +252,7 @@ def setup(
             self.logger.info("Preparing preprocessing pipeline...")
 
             # Encode the target column
-            if list(self.y.unique()) != list(range(self.y.nunique())):
+            if sorted(list(self.y.unique())) != list(range(self.y.nunique())):
```
Comment (Yard1) [inline pycaret/classification/oop.py:255]: maybe use set instead?

File: pycaret/classification/oop.py [hunk 2]
```diff
@@ -248,7 +252,7 @@ def setup(
             self.logger.info("Preparing preprocessing pipeline...")
 
             # Encode the target column
-            if self.y.dtype.kind not in "ifu":
+            if sorted(list(self.y.unique())) != list(range(self.y.nunique())):
```
Comment (Yard1) [inline pycaret/classification/oop.py:255]:
```suggestion
            y_unique = self.y.unique()
            if sorted(list(y_unique)) != list(range(len(y_unique))):
```

**Review by Yard1 — APPROVED** [review]
Looks good! One small suggestions.

**Comment by turkalpmd** [comment]
Hi there,

I am also getting the same error.

My version is  = '3.0.0.rc3'

How Fixed is this bug, sir? 


---------------------------------------------------------------------------
### I'm used this line for cluster model. 
---------------------------------------------------------------------------
tuned_model = tune_model(model,
                          supervised_target = 'Microcalcification',
                          supervised_type = 'classification',
                          supervised_estimator = 'catboost',
                          optimize = 'F1')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_183/1331734822.py in <module>
      3                          supervised_type = 'classification',
      4                          supervised_estimator = 'catboost',
----> 5                          optimize = 'F1')

/opt/conda/lib/python3.7/site-packages/pycaret/internal/utils.py in wrapper(*args, **kwargs)
    920                 if globals_d[name] is None:
    921                     raise ValueError(message)
--> 922             return func(*args, **kwargs)
    923 
    924         return wrapper

/opt/conda/lib/python3.7/site-packages/pycaret/clustering/functional.py in tune_model(model, supervised_target, supervised_type, supervised_estimator, optimize, custom_grid, fold, fit_kwargs, groups, round, verbose)
    864         groups=groups,
    865         round=round,
--> 866         verbose=verbose,
    867     )
    868 

/opt/conda/lib/python3.7/site-packages/pycaret/internal/pycaret_experiment/unsupervised_experiment.py in tune_model(self, model, supervised_target, supervised_type, supervised_estimator, optimize, custom_grid, fold, groups, ground_truth, method, fit_kwargs, round, verbose, **kwargs)
    859         if supervised_type == "classification":
    860             metrics = (
--> 861                 pycaret.containers.metrics.classification.get_all_metric_containers(
    862                     self, raise_errors=True
    863                 )

AttributeError: module 'pycaret.containers.metrics' has no attribute 'classification'

**Comment by tvdboom** [comment]
please, open a new issue with a reproducible example.

**Pull request merged**